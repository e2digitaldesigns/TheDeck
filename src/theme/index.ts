export * from "./styles/forms.styles";
export * from "./styles/scrollBars.styles";
