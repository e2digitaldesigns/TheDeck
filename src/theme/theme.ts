export const theme = {
  sizes: {
    application: {
      height: "785px",
      width: "1280px"
    },
    header: {
      height: "2rem",
      width: "100vw"
    },
    macroDeck: {
      sideBar: { width: "240px" },
      mainContent: { width: "1016px" }
    }
  },
  colors: {
    body: "black",
    border: "#111",
    accent1: "#f542e9",
    accent2: "#291697",
    accent3: "#404040",
    accent4: "#2b2b2b",
    accent5: "#1a1a1a",
    gridBg: "43, 45, 48,",
    font: "white",
    main: "#100727",
    mainBackground: "black",
    scrollBackground: "#100727",
    scrollFont: "#dedede",
    scrollHeader: "black",
    scrollHeaderFont: "white"
  }
};
