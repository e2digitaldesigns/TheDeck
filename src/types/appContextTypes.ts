export interface IntAppDataIconSelector {
  isVisible: boolean;
  icon: string;
}

export interface IntAppDataActive {
  profileId: string;
  pageId: string;
  buttonPadId: string;
  actionId: string;
}

export interface IntDropZones {
  actionList: boolean;
  buttonPads: boolean;
  sideBarProfiles: boolean;
  sideBarStyles: boolean;
  styleHeader: boolean;
}

export interface IntAppContextInterface {
  iconSelector: IntAppDataIconSelector;
  active: IntAppDataActive;
  dropZones: IntDropZones;
}

export interface IntAppContextStateInterface {
  appState: IntAppContextInterface;
  setAppState: React.Dispatch<React.SetStateAction<any>>;
}

export type IntAppData = IntAppContextStateInterface;

export enum AppStateActive {
  ProfileId = "profileId",
  PageId = "pageId",
  ButtonPadId = "buttonPadId",
  ActionId = "actionId"
}
