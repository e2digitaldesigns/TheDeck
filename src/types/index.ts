/* istanbul ignore file */
export * from "./actionType";
export * from "./aplicationTypes";
export * from "./appContextTypes";
export * from "./buttonPadGridTypes";
export * from "./dragAndDropTypes";
export * from "./globalContextType";
export * from "./globalType";
export * from "./ipcRenderTypes";
export * from "./obsTypes";
