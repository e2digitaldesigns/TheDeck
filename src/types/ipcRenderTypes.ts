export enum IpcRendererTypes {
  database = "database",
  databaseReturn = "database:return",
  header = "header",
  importSettings = "importSettings",
  importSettingsReturn = "importSettings:return",
  mdPlaySound = "MD:playSound"
}

export enum IpcRendererTypesAction {
  close = "close",
  fsToggle = "fsToggle",
  minimize = "minimize",
  loadAppData = "loadAppData",
  saveAppData = "saveAppData"
}
