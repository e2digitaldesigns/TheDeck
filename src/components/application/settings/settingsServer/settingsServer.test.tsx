import "@testing-library/jest-dom";
import { fireEvent, render, waitFor } from "@testing-library/react";
import _cloneDeep from "lodash/cloneDeep";
import SettingsServer from "./settingsServer";
import { ThemeProvider } from "styled-components";
import { theme } from "../../../../theme/theme";

import initGlobalState from "../../../../__mock_data__/globalState.json";
import { IntGlobalData } from "../../../../types";

const mockGlobalState: IntGlobalData = {
  state: _cloneDeep(initGlobalState),
  setState: jest.fn(state => state)
};

jest.mock("../../../../hooks", () => ({
  useAppData: () => ({ appState: { dropZones: { buttonPads: false } } }),
  useButton: () => ({
    activateButtonPad: jest.fn(),
    createButtonPad: jest.fn(),
    deleteButtonPad: jest.fn(),
    readButtonPad: () => mockGlobalState.state.buttonPads[0]
  }),
  useGlobalData: () => mockGlobalState,
  useProfile: () => ({
    readProfile: () => mockGlobalState.state.profiles[0]
  })
}));

const setup = () => {
  return render(
    <ThemeProvider theme={theme}>
      <SettingsServer />
    </ThemeProvider>
  );
};

describe("<SettingsServer />", () => {
  it("Should render correctly", () => {
    const wrapper = setup();
    expect(wrapper).toMatchSnapshot();
  });

  it("Should update form fields & submit", async () => {
    const wrapper = setup();
    const port = wrapper.getByTestId(
      "settings_general__port"
    ) as HTMLInputElement;
    expect(port).toBeTruthy();
    expect(port.value).toBe(mockGlobalState.state.settings.port);

    const value = "9586";
    fireEvent.change(port, { target: { value: value } });
    expect(port.value).toBe(value);

    const submit = wrapper.getByTestId("settings_general__submit");
    fireEvent.click(submit);

    await waitFor(() =>
      expect(mockGlobalState.setState).toHaveBeenCalledTimes(1)
    );
  });
});
