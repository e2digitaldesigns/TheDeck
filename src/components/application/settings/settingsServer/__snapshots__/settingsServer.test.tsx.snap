// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SettingsServer /> Should render correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-pVTFL cScoAe"
      >
        <fieldset
          class="sc-jrQzAO fFPygj"
        >
          <h2
            class="sc-kDTinF cHynfT"
          >
            Server Settings
          </h2>
          <div
            class="sc-iqseJM fbEUai"
          >
            <label
              for="settings_general__ipAddress"
            >
              IP Address:
            </label>
            <span>
              10.0.0.91
            </span>
          </div>
          <div
            class="sc-iqseJM fbEUai"
          >
            <label
              for="settings_general__port"
            >
              Port:
            </label>
            <input
              data-testid="settings_general__port"
              name="settings_general__port"
              type="text"
              value="8585"
            />
          </div>
          <div
            class="sc-iqseJM fbEUai"
          >
            <span />
            <span>
              http://10.0.0.91:8585
            </span>
          </div>
          <div
            class="sc-iqseJM fbEUai"
          >
            <span />
            <span>
              You will need to restart this application after changing the Port Number.
            </span>
          </div>
          <div
            class="sc-iqseJM fbEUai"
          >
            <span />
            <span>
              <button
                class="sc-hKwDye sc-crHmcD PLLBx houtcn"
                data-testid="settings_general__submit"
              >
                Submit
              </button>
            </span>
          </div>
        </fieldset>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-pVTFL cScoAe"
    >
      <fieldset
        class="sc-jrQzAO fFPygj"
      >
        <h2
          class="sc-kDTinF cHynfT"
        >
          Server Settings
        </h2>
        <div
          class="sc-iqseJM fbEUai"
        >
          <label
            for="settings_general__ipAddress"
          >
            IP Address:
          </label>
          <span>
            10.0.0.91
          </span>
        </div>
        <div
          class="sc-iqseJM fbEUai"
        >
          <label
            for="settings_general__port"
          >
            Port:
          </label>
          <input
            data-testid="settings_general__port"
            name="settings_general__port"
            type="text"
            value="8585"
          />
        </div>
        <div
          class="sc-iqseJM fbEUai"
        >
          <span />
          <span>
            http://10.0.0.91:8585
          </span>
        </div>
        <div
          class="sc-iqseJM fbEUai"
        >
          <span />
          <span>
            You will need to restart this application after changing the Port Number.
          </span>
        </div>
        <div
          class="sc-iqseJM fbEUai"
        >
          <span />
          <span>
            <button
              class="sc-hKwDye sc-crHmcD PLLBx houtcn"
              data-testid="settings_general__submit"
            >
              Submit
            </button>
          </span>
        </div>
      </fieldset>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
